system/services/assetManager/AssetManagerService.cfc

```diff
@@ -972,6 +972,13 @@ component displayName="AssetManager Service" {
 		var filename        = arguments.assetId & ( Len( Trim( arguments.versionId ) ) ? ".#arguments.versionId#" : "" ) & ".#fileext#";
 		var derivativeSlug  = ReReplace( arguments.derivativeName, "\W", "_", "all" ) & "_" & signature;
 		var storagePath     = "/derivatives/#derivativeSlug#/#filename#";
+		var assetDimension  = getAsset( id=arguments.assetId, throwOnMissing=true, selectFields=[ "height",'width' ] );
+
+		var assetProperty               = {};
+		assetProperty.height            = assetDimension.height ?: 0;
+		assetProperty.width             = assetDimension.width  ?: 0;
+		assetProperty.sourceFile        = expandPath( '/uploads/assets' & asset.storage_path );
```

Comment:
This is not correct. We cannot know that assets will live at that path due to the possibility of multiple configured storage providers in the asset manager.

We need to do something like:

```
assetManagerService.getPhysicalPath( assetId );
```

This new function would need to get the storage provider for the asset and call a `getPhysicalPath()` method on the asset provider (this would also be a new method for the storage provider interface). For the file system storage provider, this would be a simple case of returning the file path combined with the storage path. For something like an S3 storage provider, we might need to implement something like a local cache directory for pulling down the file locally when we need access to a local file.